__tests__/server.test.js

```diff
@@ -266,4 +258,77 @@ describe("backend API project", () => {
+      return request(app)
+        .patch("/api/articles/1")
+        .send({
+          inc_votes: 1,
+        })
+        .expect(201)
+        .then(({ body }) => {
+          expect(typeof body.article).toBe("object");
+          expect(body.article).toHaveProperty("author", expect.any(String));
+          expect(body.article).toHaveProperty("title", expect.any(String));
+          expect(body.article).toHaveProperty("article_id", expect.any(Number));
+          expect(body.article).toHaveProperty("body", expect.any(String));
+          expect(body.article).toHaveProperty("topic", expect.any(String));
+          expect(body.article).toHaveProperty("created_at", expect.any(String));
+          expect(body.article).toHaveProperty("votes", 101);
+          expect(body.article).toHaveProperty(
+            "article_img_url",
+            expect.any(String)
+          );
+        });
+    });
+    test("that a PATCH request responds with the updated article when given a correct body subtracting 100", () => {
+      return request(app)
+        .patch("/api/articles/1")
+        .send({
+          inc_votes: -100,
+        })
+        .expect(201)
+        .then(({ body }) => {
+          expect(typeof body.article).toBe("object");
+          expect(body.article).toHaveProperty("author", expect.any(String));
+          expect(body.article).toHaveProperty("title", expect.any(String));
+          expect(body.article).toHaveProperty("article_id", expect.any(Number));
+          expect(body.article).toHaveProperty("body", expect.any(String));
+          expect(body.article).toHaveProperty("topic", expect.any(String));
+          expect(body.article).toHaveProperty("created_at", expect.any(String));
+          expect(body.article).toHaveProperty("votes", 0);
+          expect(body.article).toHaveProperty(
+            "article_img_url",
+            expect.any(String)
+          );
+        });
+    });
+    test("that a PATCH request with an invalid body type returns 400 Bad request", () => {
+      return request(app)
+        .patch("/api/articles/badrequest")
+        .send({ inc_votes: 1 })
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Bad request");
+        });
+    });
+    test("that a patch request with a valid body that would reduce the votes below 0 returns 406 Not Acceptable", () => {
+      return request(app)
+        .patch("/api/articles/1")
+        .send({ inc_votes: 1000 })
+        .expect(406)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Not Acceptable");
+        });
+    });
+    test("that a PATCH request with a valid body and value to a article that doesnt exist returns 400 Bad request", () => {
+      return request(app)
+        .patch("/api/articles/123")
+        .send({ inc_votes: 1 })
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Bad request");
+        });
+    });
+  });
```

Comment:
To clarify, your tests should ideally be as follows
Status 200, updated single article object.(increase votes)
Status 200, updates single article object (decrease votes)
Status 400, invalid ID, e.g. string of "not-an-id"
Status 404, non existent ID, e.g. 0 or 9999
Status 400, incorrect body, e.g. inc_votes property is not a number
(Optional) Status 200, if inc_votes is missing, should not update article and return original single article object (this may be included as a 400 but that could make for a brittle test suite)